keras_cv/models/object_detection/yolo_v8/yolo_v8_detector.py

```diff
@@ -0,0 +1,636 @@
+            a Non-Max Suppression for box pruning.
+
+    Examples:
+    ```python
+    images = tf.ones(shape=(1, 512, 512, 3))
+    labels = {
+        "boxes": [
+            [
+                [0, 0, 100, 100],
+                [100, 100, 200, 200],
+                [300, 300, 100, 100],
+            ]
+        ],
+        "classes": [[1, 1, 1]],
+    }
+    model = keras_cv.models.YOLOV8Detector(
+        num_classes=20,
+        bounding_box_format="xywh",
+        backbone=keras_cv.models.YOLOV8Backbone.from_preset(
+            "yolov8_m_coco"
+        ),
+        fpn_depth=2.
+    )
+
+    # Evaluate model
+    model(images)
+
+    # Get predictions using the model
+    model.predict(images)
+
+    # Train model
+    model.compile(
+        classification_loss='binary_crossentropy',
+        box_loss='iou',
+        optimizer=tf.optimizers.SGD(global_clipnorm=10.0),
+        jit_compile=False,
+    )
+    model.fit(images, labels)
+    ```
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        num_classes,
+        bounding_box_format,
+        backbone,
+        fpn_depth,
+        label_encoder=None,
+        prediction_decoder=None,
+        **kwargs,
+    ):
+        extractor_levels = [3, 4, 5]
+        extractor_layer_names = [
+            backbone.pyramid_level_inputs[i] for i in extractor_levels
+        ]
+        feature_extractor = get_feature_extractor(
+            backbone, extractor_layer_names, extractor_levels
+        )
+
+        images = layers.Input(feature_extractor.input_shape[1:])
+        features = list(feature_extractor(images).values())
+
+        fpn_features = apply_path_aggregation_fpn(
+            features, depth=fpn_depth, name="pa_fpn"
+        )
+
+        outputs = apply_yolov8_head(
+            fpn_features,
+            num_classes,
+        )
+
+        # To make loss metrics pretty, we use a no-op layer with a good name.
+        boxes = keras.layers.Concatenate(axis=1, name="box")([outputs["boxes"]])
+        scores = keras.layers.Concatenate(axis=1, name="class")(
+            [outputs["classes"]]
+        )
+
+        outputs = {"boxes": boxes, "classes": scores}
+        super().__init__(inputs=images, outputs=outputs, **kwargs)
+
+        self.bounding_box_format = bounding_box_format
+        self.prediction_decoder = (
+            prediction_decoder
+            or keras_cv.layers.MultiClassNonMaxSuppression(
+                bounding_box_format=bounding_box_format,
+                from_logits=False,
+                confidence_threshold=0.2,
+                iou_threshold=0.7,
+            )
+        )
+        self.backbone = backbone
+        self.fpn_depth = fpn_depth
+        self.num_classes = num_classes
+        self.label_encoder = label_encoder or YOLOV8LabelEncoder(
+            num_classes=num_classes
+        )
+
+    def compile(
+        self,
+        box_loss,
+        classification_loss,
```

Comment:
What could be the complication if it's followed as 

```python
Model.compile(
    optimizer=...,
    loss={
       "box_loss": box_loss,
       "classification_loss": classification_loss
    },
    metrics=...,
    loss_weights={
       "box_loss_weight": box_loss_weight,
       "classification_loss_weight": classification_loss_weight
    },
    weighted_metrics=...,

)
```

Reply:
I recall that we considered this API when we originally designed the `RetinaNet.compile` signature, but I can't recall why we decided against it.

Maybe @LukeWood knows?

Reply:
if both arguments are required it is weird to nest them in a dictionary.  Dictionary nested required arguments are an antipattern IMO.  You lose autocomplete, have to write out that ugly dict, etc.

Reply:
I see. It might need to be reconsidered otherwise existing API would be needed to rewrite for many categories (classification, detection, segmentation, etc). 

https://keras.io/api/models/model_training_apis/
```python
Model.compile(
    optimizer="rmsprop",
    loss=None,
    metrics=None,
    loss_weights=None,
    weighted_metrics=None,
    run_eagerly=None,
    steps_per_execution=None,
    jit_compile=None,
    **kwargs
)
```